src/caf/toolkit/cost_utils.py

```diff
@@ -6,98 +6,116 @@
 
 
 # # # CLASSES # # #
-@dataclass(config={"arbitrary_types_allowed": True})  # type: ignore # pylint: disable=used-before-assignment, unexpected-keyword-arg
 class CostDistribution:
     """Distribution of cost values between variable bounds.
 
-    Attributes
+    Alternate constructors are available in the See Also section
+
+    Parameters
     ----------
     df:
-        The raw Pandas DataFrame containing the cost distribution data.
+        A DataFrame containing the binned cost distribution data. Must have columns
+        named: `min_col`, `max_col`, `avg_col`, `trips_col`.
 
-    min_vals:
-        Minimum values of the cost distribution bin edges.
+    min_col:
+        The name of the columns in `df` that contains the lower bin edge
+        value for each row.
 
-    max_vals:
-        Maximum values of the cost distribution bin edges.
+    max_col:
+        The name of the columns in `df` that contains the upper bin edge
+        value for each row.
 
-    bin_edges:
-        Bin edges for the cost distribution.
-
-    avg_vals:
-        Average values for each of the cost distribution bins.
+    avg_col:
+        The name of the columns in `df` that contains the centre of the bin
 
-    trip_vals:
-        Trip values for each of the cost distribution bins.
+    trips_col:
+        The name of the columns in `df` that contains the value for each
+        row.
 
-    band_share_vals:
-        Band share values for each of the cost distribution bins.
+    weighted_avg_col:
+        The name of the columns in `df` that contains the weighted average
+        value for each row. If available, this is different from `avg_col`
+        as it takes into account this distribution of values within each
+        bound when calculating averages.
 
-    weighted_avg_vals:
-        Weighted average values for each of the cost distribution bins.
+    See Also
+    --------
+    :func:`from_data`
+    :func:`from_data_no_bins`
+    :func:`from_file`
     """
 
-    df: pd.DataFrame
-
-    # Default arguments
-    min_col: str = "min"
-    max_col: str = "max"
-    avg_col: str = "ave"
-    trips_col: str = "trips"
-    weighted_avg_col: Optional[str] = "weighted_ave"
-
     # Ideas
     units: str = "km"
 
-    @pydantic.model_validator(mode="after")
-    def check_df_col_names(self) -> CostDistribution:
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        min_col: str = "min",
+        max_col: str = "max",
+        avg_col: str = "avg",
+        trips_col: str = "trips",
+        weighted_avg_col: Optional[str] = None,
+    ):
+        self.df = df
+
+        # Keep as private. These shouldn't be needed outside of this class
+        self.__min_col = min_col
+        self.__max_col = max_col
+        self.__avg_col = avg_col
+        self.__trips_col = trips_col
+
+        if weighted_avg_col is None:
+            weighted_avg_col = "weighted_avg"
+        self.__weighted_avg_col = weighted_avg_col
```

Comment:
This could just be the parameters default value since it doesn't do anything special if given None